src/poprox_recommender/recommenders/__init__.py

```diff
@@ -29,10 +36,12 @@ def select_articles(
     Select articles with default recommender configuration.  It returns a
     pipeline state whose ``default`` is the final list of recommendations.
     """
+    name = None
     if pipeline_params and "pipeline" in pipeline_params:
         name = pipeline_params["pipeline"]
-    else:
-        name = "nrms"
+
+    if name is None:
+        name = default_pipeline()
```

Comment:
This is the key part that fixes the bug. The rest is a test + better logic for defining what the default pipeline is.